src/components/tx-icon.tsx

```diff
@@ -7,27 +7,47 @@ import {
   FiAlertOctagon as IconMoodSad,
   FiPlus as IconPlus,
 } from 'react-icons/fi';
-import { Box, BoxProps, Circle, color, DynamicColorCircle, StxNexus } from '@stacks/ui';
+import {
+  Box,
+  BoxProps,
+  Circle,
+  color,
+  ColorsStringLiteral,
+  DynamicColorCircle,
+  StxNexus,
+} from '@stacks/ui';
 import FunctionIcon from 'mdi-react/FunctionIcon';
-import type { MempoolTransaction, Transaction } from '@stacks/stacks-blockchain-api-types';
 import { useWallet } from '@common/hooks/use-wallet';
-
-type Tx = MempoolTransaction | Transaction;
+import { MicroblockIcon } from '@components/icons/microblock';
+import { Tx, Statuses, statusFromTx } from '@common/api/transactions';
 
 interface TypeIconWrapperProps extends BoxProps {
   icon: React.FC<any>;
-  status: Tx['tx_status'];
+  bg: any;
 }
 
-const TypeIconWrapper: React.FC<TypeIconWrapperProps> = ({ status, icon: Icon, ...rest }) => (
+type statusColorMap = {
+  [key in Statuses]: ColorsStringLiteral;
+};
```

Comment:
Can also be described as a record, also should use pascal casing for types:

```suggestion
type StatusColorMap = Record<Statuses, ColorsStringLiteral>;
```

Reply:
Yes indeed this is cleaner.